[Feature] Add torch_npu optimizer (#1079)
open-mmlab/mmengine: docs/en/common_usage/speed_up_training.md

```{warning}
`torch.compile` is still under development by PyTorch team. Some models may fail compilation. If you encounter errors during compilation, you can refer to [PyTorch Dynamo FAQ](https://pytorch.org/docs/2.0/dynamo/faq.html) for quick fix, or [TorchDynamo Troubleshooting](https://pytorch.org/docs/2.0/dynamo/troubleshooting.html) to post an issue in PyTorch.
```

> ## Faster Optimizers
> 
> If Ascend's equipment is used, Ascend's optimizer can be used to reduce the training time of the model. The optimizers supported by Ascend devices are as follows
> 
> - NpuFusedAdadelta
> - NpuFusedAdam
> - NpuFusedAdamP
> - NpuFusedAdamW
> - NpuFusedBertAdam
> - NpuFusedLamb
> - NpuFusedRMSprop
> - NpuFusedRMSpropTF
> - NpuFusedSGD
> 
> The usage method is the same as the native optimizer, please refer to [use of optimizers](../tutorials/optim_wrapper.md#configure-the-optimwapper-in-runner)

Reviewer: ```suggestion
## Using faster Optimizers

If Ascend devices are used, you can use the Ascend optimizers to shorten the training time of the model. The optimizers supported by Ascend devices are as follows:

- NpuFusedAdadelta
- NpuFusedAdam
- NpuFusedAdamP
- NpuFusedAdamW
- NpuFusedBertAdam
- NpuFusedLamb
- NpuFusedRMSprop
- NpuFusedRMSpropTF
- NpuFusedSGD

The usage is the same as native optimizers, and you can refer to [Using Optimizers](../tutorials/optim_wrapper.md#configure-the-optimwapper-in-runner) for more information.
```

```{warning}
`torch.compile` is still under development by PyTorch team. Some models may fail compilation. If you encounter errors during compilation, you can refer to [PyTorch Dynamo FAQ](https://pytorch.org/docs/2.0/dynamo/faq.html) for quick fix, or [TorchDynamo Troubleshooting](https://pytorch.org/docs/2.0/dynamo/troubleshooting.html) to post an issue in PyTorch.
```

## Faster Optimizers

If Ascend's equipment is used, Ascend's optimizer can be used to reduce the training time of the model. The optimizers supported by Ascend devices are as follows

> ```
> NpuFusedAdadelta
> NpuFusedAdam
> NpuFusedAdamP
> NpuFusedAdamW
> NpuFusedBertAdam
> NpuFusedLamb
> NpuFusedRMSprop
> NpuFusedRMSpropTF
> NpuFusedSGD
> ```

Reviewer: ```suggestion

- NpuFusedAdadelta
- NpuFusedAdam
- NpuFusedAdamP
- NpuFusedAdamW
- NpuFusedBertAdam
- NpuFusedLamb
- NpuFusedRMSprop
- NpuFusedRMSpropTF
- NpuFusedSGD

```

> The usage method is the same as the native optimizer, and the more detailed usage method in MMEngine can be referred to [optimizers](https://mmengine.readthedocs.io/zh_CN/latest/tutorials/optim_wrapper.html?highlight=%E4%BC%98%E5%8C%96%E5%99%A8).

Reviewer: Please also update the en doc.
